nas/auth.go

```diff
@@ -38,22 +38,28 @@ func handleAuthRequest(moduleName string, w http.ResponseWriter, r *http.Request
 			continue
 		}
 
-		parsed, err := common.Base64DwcEncoding.DecodeString(values[0])
-		if err != nil {
-			logging.Error(moduleName, "Invalid POST form value:", aurora.Cyan(key).String()+":", aurora.Cyan(values[0]))
-			replyHTTPError(w, 400, "400 Bad Request")
-			return
-		}
-
-		value := string(parsed)
+		var value string
+		if !strings.HasPrefix(key, "_") {
+			parsed, err := common.Base64DwcEncoding.DecodeString(values[0])
+			if err != nil {
+				logging.Error(moduleName, "Invalid POST form value:", aurora.Cyan(key).String()+":", aurora.Cyan(values[0]))
+				replyHTTPError(w, 400, "400 Bad Request")
+				return
+			}
 
-		if key == "ingamesn" {
-			// Special handling required for the UTF-16 string
-			var utf16String []uint16
-			for i := 0; i < len(parsed)/2; i++ {
-				utf16String = append(utf16String, binary.BigEndian.Uint16(parsed[i*2:i*2+2]))
+			if key == "ingamesn" {
+				// Special handling required for the UTF-16 string
+				var utf16String []uint16
+				for i := 0; i < len(parsed)/2; i++ {
+					utf16String = append(utf16String, binary.BigEndian.Uint16(parsed[i*2:i*2+2]))
```

Comment:
```go
for i := 0; i < len(parsed); i+=2 {
    utf16String = append(utf16String, binary.BigEndian.Uint16(parsed[i:]))
}
```

Would be preferred rather than what is here. You don't need to pass exactly 2 bytes to `binary.BigEndian.Uint16`, the function will convert the first 2 bytes in the slice to a u16.